src/util/util/Email.ts

```diff
@@ -41,3 +45,56 @@ export function adjustEmail(email?: string): string | undefined {
 
 	return email;
 }
+
+export const Email: {
+	transporter: Transporter | null;
+	init: () => Promise<void>;
+	sendVerificationEmail: (id: string, email: string) => Promise<any>;
+} = {
+	transporter: null,
+	init: async function () {
+		const { host, port, secure, username, password } = Config.get().smtp;
+		if (!host || !port || !secure || !username || !password) return;
+		console.log(`[SMTP] connect: ${host}`);
+		this.transporter = nodemailer.createTransport({
+			host,
+			port,
+			secure,
+			auth: {
+				user: username,
+				pass: password,
+			},
+		});
+
+		await this.transporter.verify((error, _) => {
+			if (error) {
+				console.error(`[SMTP] error: ${error}`);
+				this.transporter?.close();
+				this.transporter = null;
+				return;
+			}
+			console.log(`[SMTP] Ready`);
+		});
+	},
+	sendVerificationEmail: async function (
+		id: string,
+		email: string,
+	): Promise<any> {
+		if (!this.transporter) return;
+		const token = (await generateToken(id, email)) as string;
+		const instanceUrl =
+			Config.get().general.frontPage || "http://localhost:3001";
+		const link = `${instanceUrl}/verify#token=${token}`;
+		const message = {
+			from:
+				Config.get().general.correspondenceEmail || "noreply@localhost",
+			to: email,
+			subject: `Verify Email Address for ${
+				Config.get().general.instanceName
+			}`,
+			html: `Please verify your email address by clicking the following link: <a href="${link}">Verify Email</a>`,
```

Comment:
I wonder how hard it would be to make it load the html content from a file to make it configurable by the owner of an instance.

Reply:
should be pretty easy, was thinking just load the template, and replace placeholders